pallets/restricted-tokens/src/impl_fungibles.rs

```diff
@@ -151,12 +151,15 @@ impl<T: Config> InspectHold<T::AccountId> for Pallet<T> {
 		if asset == T::NativeToken::get() {
 			<Pallet<T> as fungible::InspectHold<T::AccountId>>::can_hold(who, amount)
 		} else {
+			let can_hold =
+				<T::Fungibles as InspectHold<T::AccountId>>::can_hold(asset, who, amount);
+
 			T::PreFungiblesInspectHold::check(FungiblesInspectHoldEffects::CanHold(
 				asset,
 				who.clone(),
 				amount,
-				<T::Fungibles as InspectHold<T::AccountId>>::can_hold(asset, who, amount),
-			))
+				can_hold,
+			)) && can_hold
```

Comment:
Much thanks for fixing! Questioning the thoroughness of our existing `restricted-tokens` test if this was uncaught but that's out of scope for this PR.

Reply:
Yup, I also ran the tests locally expecting a failure but everything is green...